pkg/kubelet/kubelet.go

```diff
@@ -1129,12 +1129,18 @@ func (kl *Kubelet) syncPod(pod *api.Pod, mirrorPod *api.Pod, runningPod kubecont
 	}
 	kl.volumeManager.SetVolumes(pod.UID, podVolumes)
 
-	podStatus, err := kl.generatePodStatus(pod)
-	if err != nil {
-		glog.Errorf("Unable to get status for pod %q (uid %q): %v", podFullName, uid, err)
-		return err
+	var podStatus api.PodStatus
+	if _, ok := kl.statusManager.GetPodStatus(podFullName); ok {
+		var err error
+		podStatus, err = kl.generatePodStatus(pod)
+		if err != nil {
+			glog.Errorf("Unable to get status for pod %q (uid %q): %v", podFullName, uid, err)
+			return err
+		}
+	} else {
+		glog.V(3).Infof("New pod %v, using default pod status", podFullName)
```

Comment:
Don't think this is accurate. Pod statuses in `statusManager` cache may get deleted for some reasons. E.g., (1) if the mirror pod hasn't been created yet, the corresponding status of the static pod will not be kept in the cache; (2) pod status gets deleted if status update (to the apiserver) fails for any reason, so that it will get sent again in the next sync.

It'd be safer to pass the sync type (creation, update, etc) to the syncPod function.